sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,462 @@
+        else:
+            self.wavelength_range = self.dataframe['94']['wave']
+
+
+            # Notes: I'm trying to define self.stuffs as cleanly as possible - Any input appreciated as this is still new to me
+            #          I saw that def __repr__(self): can be used to define self.things but I'm not sure on how..?
+
+    def aia_inst_genx_to_dataframe(self, path_to_file, save=False):
+        """
+        This definition reads the instrument file aia_V6_all_fullinst, which was obtained from ssw_aia_response_data inside
+        ssw_aia_response_genx.tar.gz (an output file saved from SolarSoft Ware (SSW)). It will extract the instrument data and save
+        it a dataframe file for easier access.
+
+         Parameters
+        ----------
+        path_to_file : string, the path location that leads to ssw_aia_response_data/aia_V6_all_fullinst.
+
+        :param save: bool, if True, will save a dataframe containing all of the extracted instrument data to
+
+        Returns
+        -------
+        :returns self.dataframe: dataframe, stores the dictionary for easy access. Each column lists the channel information and each row is a different peroperty from aia_inst_genx.
+
+        :returns 'channel_properties.csv', dataframe with wavelength centers for each channel is a column and listed in the column are the properties from the is returned where it has keys of the properties of the channel
+
+        Notes:
+        np.recarray store information with shape(1,0) and are quite nested.
+
+        """
+
+        # access np.recarray from .genx file
+        ssw_array = readsav(path_to_file)
+        data = ssw_array['data']
+
+        # store in dataframe
+        df = pd.DataFrame()
+
+        # pick out instrument files inside np.recarray
+        for name in data.dtype.names:
+
+            # eg.(A94_THICK_FULL or A94_FULL or A94_THICK_FILE)
+            # TODO: Fix to read in more than just this file to get all desired channels
+            if name.startswith('A') and name.endswith('THICK_FULL'):
+
+                # target number in filename that matches wavelength_center
+                start = name.find('A')
+                end = name.find('_T')
+                wavelength = name[start + 1:end]
+
+                # match the properties in the rec.array to desired properties
+                variables = set(data[name][0].dtype.fields.keys()) & set(self.properties)
+                assert len(variables) == len(self.properties)
+
+                # then iterate through the properties to fill a dataframe with rec.array information
+                for property in variables:
+                    try:
+                        df.loc[property, wavelength] = data[name][0][property][0]
+
+                    # contam and crossarea are sequences so these don't have to be unpacked as much
+                    except ValueError:
+                        df.loc[property, wavelength] = data[name][0][property]
```

Comment:
Yes, that was what I meant, I don't think either of them would produce a `ValueError`. Perhaps a better way to do this would be with an if/else rather than a try/except.